win32ss/user/ntuser/ntstubs.c

```diff
@@ -416,12 +416,56 @@ NtUserYieldTask(VOID)
    return 0;
 }
 
+PIMC APIENTRY UserCreateInputContext(ULONG_PTR dwClientImcData)
+{
+    PIMC pIMC;
+    PTHREADINFO pti = PsGetCurrentThreadWin32Thread();
+    PDESKTOP pdesk = pti->rpdesk;
+
+    if (!IS_IMM_MODE() || (pti->TIF_flags & TIF_DISABLEIME) || !pdesk)
+        return NULL;
+
+    if (dwClientImcData && !pti->spDefaultImc)
+        return NULL;
+
+    pIMC = UserCreateObject(gHandleTable, pdesk, pti, NULL, TYPE_INPUTCONTEXT, sizeof(IMC));
+    if (!pIMC)
+        return NULL;
+
+    if (dwClientImcData)
```

Comment:
Your code misses some informative comments, like for example, why do we need to link the pIMC in different ways when that client data is non-zero or not....

Reply:
Added comments.